test/e2e/tests/commons/sinspqa/sinsp.py

```diff
@@ -0,0 +1,218 @@
+        """
+        decoded_log = raw_log.decode("ascii").strip()
+        split_log = decoded_log.split(" ")
+        return datetime.strptime(split_log[0][:-4], "%Y-%m-%dT%H:%M:%S.%f"), " ".join(split_log[1:])
+
+
+class SinspFieldTypes(Enum):
+    STRING = 0
+    REGEX = 1
+
+
+class SinspField:
+    """
+    Stores the value expected in a field output by sinsp-example.
+    """
+    def __init__(self, value, value_type=SinspFieldTypes.STRING):
+        self.value_type = value_type
+
+        if self.value_type == SinspFieldTypes.REGEX:
+            self.value = re.compile(value)
+        else:
+            self.value = value
+
+    def compare(self, other: str) -> bool:
+        if self.value_type == SinspFieldTypes.REGEX:
+            return self.value.match(other)
+
+        return self.value == other
+
+    def __repr__(self):
+        if self.value_type == SinspFieldTypes.REGEX:
+            return f"r'{self.value.pattern}'"
+        else:
+            return self.value
+
+    def numeric_field():
+        return SinspField(r'^\d+$', SinspFieldTypes.REGEX)
+
+    def regex_field(regex: str):
+        return SinspField(regex, SinspFieldTypes.REGEX)
+
+
+def parse_log(log):
+    """
+    Parses a log line from the `sinsp-example` binary.
+
+    Parameters:
+        log (str): A string holding a single log line from `sinsp-example``
+
+    Returns:
+        A dictionary holding all the captured values for the event.
+    """
+    return json.loads(log)
+
+
+def validate_event(expected_fields, event):
+    """
+    Checks all `expected_fields` are in the `event`
+
+    Parameters:
+        expected_fields (dict): A dictionary holding the values expected in the event.
+        event (dict): A sinsp event parsed by calling `parse_log`
+
+    Returns:
+        True if all `expected_fields` are in the event and have matching values, False otherwise.
+    """
+    for k in expected_fields:
+        if k not in event:
+            return False
+
+        expected = expected_fields[k]
+
+        if isinstance(expected, str) or expected is None:
+            if expected == event[k]:
+                continue
+            return False
+
+        if not expected.compare(str(event[k])):
+            return False
+
+    return True
+
+
+def assert_events(expected_events, container):
+    """
+    Takes a list of dictionionaries describing the events we want to receive
+    from a sinsp-example container and the reads events from the provided
+    container handle until either all events are found or a timeout occurs.
+
+    Parameters:
+        expected_fields (dict): A dictionary holding the values expected in the event.
+        container (docker.Container): A container object to stream logs from.
+    """
+
+    reader = SinspStreamer(container)
+
+    for event in expected_events:
+        success = False
+
+        for log in reader.read():
+            if not log:
+                continue
+
+            if validate_event(event, parse_log(log)):
```

Comment:
So having built and run the tests I did run into this problem - whereby `sinsp-example` crashed because of a missing `GLIBCXX` version and I got a `JSONDecodeError`:

```
/usr/local/lib/python3.7/dist-packages/sinspqa/sinsp.py:161: in assert_events
    if validate_event(event, parse_log(log)):
/usr/local/lib/python3.7/dist-packages/sinspqa/sinsp.py:110: in parse_log
    return json.loads(log)
/usr/lib/python3.7/json/__init__.py:348: in loads
    return _default_decoder.decode(s)
/usr/lib/python3.7/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <json.decoder.JSONDecoder object at 0x7fe9cfbb73c8>
s = "sinsp-example: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by sinsp-example)"
```

Admittedly it's pretty clear in this case but I think the point still stands that we could add some context about the error

Reply:
I admit I've thought about this a few times and never got around to actually implementing it, this is as good a time as any other 🙂